packages/lodestar/src/network/gossip/gossip.ts

```diff
@@ -123,24 +126,39 @@ export class Gossip extends (EventEmitter as { new(): GossipEventEmitter }) impl
     subnet: number|string,
     callback?: (attestation: {attestation: Attestation; subnet: number}) => void
   ): void {
-    this.subscribe(forkDigest, GossipEvent.ATTESTATION_SUBNET, callback, new Map([["subnet", subnet.toString()]]));
+    const subnetNum: number = (typeof subnet === "string")? parseInt(subnet) : subnet as number;
+    this.subscribe(forkDigest, subnetNum, callback, new Map([["subnet", subnet.toString()]]));
```

Comment:
```suggestion
    this.subscribe(forkDigest, GossipEvent.ATTESTATION_SUBNET + "_" +subnetNum, callback, new Map([["subnet", subnet.toString()]]));
```
because event name as just number is weird

Reply:
I failed to do that. I don't want to declare IGossipEvents like:
```
"committee_index_beacon_attestation_0": (attestationSubnet: {attestation: Attestation; subnet: number}) => void;
"committee_index_beacon_attestation_1": (attestationSubnet: {attestation: Attestation; subnet: number}) => void;
...
62 more
```
If I simply do `[key: string]: ...` I got compilation error due to conflict to other events so I have to do `[key:number]` to avoid the conflict.
any ideas to declare attestation subnet events in `IGossipEvents` as a string pattern?

Reply:
the only way I can do is to have a separate event emitter for attestation subnet, please have a look.

Reply:
That's a pickle, maybe the best way is to "ts-ignore" where it complains since dynamic event name is not supported in strict event emitter and event that is just number is wierd as heck